detectors/node/opentelemetry-resource-detector-alibaba-cloud/test/detectors/AlibabaCloudEcsDetector.test.ts

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as nock from 'nock';
+import * as assert from 'assert';
+import { Resource } from '@opentelemetry/resources';
+import { alibabaCloudEcsDetector } from '../../src';
+import {
+  assertCloudResource,
+  assertHostResource,
+} from '@opentelemetry/test-utils';
+
+const ALIYUN_HOST =
+  'http://' + alibabaCloudEcsDetector.ALIBABA_CLOUD_IDMS_ENDPOINT;
+const ALIYUN_IDENTITY_PATH =
+  alibabaCloudEcsDetector.ALIBABA_CLOUD_INSTANCE_IDENTITY_DOCUMENT_PATH;
+const ALIYUN_HOST_PATH =
+  alibabaCloudEcsDetector.ALIBABA_CLOUD_INSTANCE_HOST_DOCUMENT_PATH;
+
+const mockedIdentityResponse = {
+  'image-id': 'my-image-id',
+  'instance-id': 'my-instance-id',
+  'instance-type': 'my-instance-type',
+  mac: 'my-mac',
+  'owner-account-id': 'my-owner-account-id',
+  'private-ipv4': 'my-private-ipv4',
+  'region-id': 'my-region-id',
+  'serial-number': 'my-serial-number',
+  'zone-id': 'my-zone-id',
+};
+const mockedHostResponse = 'my-hostname';
+
+describe('alibabaCloudEcsDetector', () => {
+  beforeEach(() => {
+    nock.disableNetConnect();
+    nock.cleanAll();
+  });
+
+  afterEach(() => {
+    nock.enableNetConnect();
+  });
+
+  describe('with successful request', () => {
+    it('should return alibaba cloud ecs instance resource', async () => {
+      const scope = nock(ALIYUN_HOST)
+        .persist()
+        .get(ALIYUN_IDENTITY_PATH)
+        .reply(200, () => mockedIdentityResponse)
+        .get(ALIYUN_HOST_PATH)
+        .reply(200, () => mockedHostResponse);
+
+      const resource: Resource = await alibabaCloudEcsDetector.detect();
+
+      scope.done();
+
+      assert.ok(resource);
+
+      assertCloudResource(resource, {
+        provider: 'alibaba_cloud',
```

Comment:
Is there a reason why you don't use semconv constants here?

Reply:
I was thinking that the test should be plain and easy to read. And it's following how the detector aws package is being tested.

Reply:
Fine for me.
Test will fail if semantic conventions are changed but I assume this will usually not happen it it would break quite a lot.